Docs: Improve v8 upgrade notes for SQL data sources (#38792)
grafana/grafana: docs/sources/installation/upgrading.md

For any existing panels/visualizations using a _Time series_ query, where the time column is only needed for filtering the time range, for example, using the bar gauge or pie chart panel, we recommend that you use a _Table query_ instead and exclude the time column as a field in the response.
Refer to this [issue comment](https://github.com/grafana/grafana/issues/35534#issuecomment-861519658) for detailed instructions and workarounds.

#### Prefix added to series names

> Given you have a query similar to below, where there's a time value and a numeric value selected together with a string value that's not named _metric_. Then you might notice that the graph panel renders series names as `value <hostname>` rather than just `<hostname>` which was the case before Grafana 8.

Reviewer: ```suggestion
When you have a query where there's a time value and a numeric value selected together with a string value that's not named _metric_, the graph panel renders series names as `value <hostname>` rather than just `<hostname>` which was the case before Grafana 8.
```

```sql
SELECT
  $__timeGroup("createdAt",'10m'),
  avg(value) as "value",
  hostname
FROM grafana_metric
WHERE $__timeFilter("createdAt")
GROUP BY time, hostname
ORDER BY time
```

> There's two possible workarounds to resolve this problem:

Reviewer: ```suggestion
There are two possible workarounds to resolve this problem:
```

> 1. Starting from Grafana v8.0.3 you should alias the string column selected as `metric`, i.e. `hostname as metric`.

Reviewer: ```suggestion
1. In Grafana v8.0.3, use alias the string column selected as `metric`. for example, `hostname as metric`.
```

Author: ```suggestion
1. In Grafana v8.0.3, use an alias of the string column selected as `metric`. for example, `hostname as metric`.
```

> 2. Use the [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}) to format the alias. Given example query above you would use `${__field.labels.hostname}`.

Reviewer: ```suggestion
2. Use the [Standard field options/Display name]({{< relref "../panels/standard-options.md#display-name" >}}) to format the alias. For the preceeding example query, you would use `${__field.labels.hostname}` option.
```

> For more information please refer to the documentation of [Postgres]({{< relref "../datasources/postgres.md#time-series-queries" >}}), [MySQL]({{< relref "../datasources/mysql.md#time-series-queries" >}}), [Microsoft SQL Server]({{< relref "../datasources/mssql.md#time-series-queries" >}}).

Reviewer: ```suggestion
For more information, refer to the our relational databases documentation of [Postgres]({{< relref "../datasources/postgres.md#time-series-queries" >}}), [MySQL]({{< relref "../datasources/mysql.md#time-series-queries" >}}), [Microsoft SQL Server]({{< relref "../datasources/mssql.md#time-series-queries" >}}).
```

## Upgrading to v8.1

### Use of unencrypted passwords for data sources no longer supported

As of Grafana v8.1, we no longer support unencrypted storage of passwords and basic auth passwords.

> > **Note":** Since Grafana v6.2, new or updated data sources store passwords and basic auth passwords encrypted. See [upgrade note]({{< relref "#ensure-encryption-of-data-source-secrets" >}}) for more information. However, unencrypted passwords and basic auth passwords were also allowed.

Reviewer: ```suggestion
> **Note":** In Grafana v6.2.x stored passwords of new or updated data sources and basic auth passwords encrypted. See [upgrade note]({{< relref "#ensure-encryption-of-data-source-secrets" >}}) for more information. However, unencrypted passwords and basic auth passwords were also allowed.
```
